src/View/Helper/PaginatorHelper.php

```diff
@@ -567,20 +567,18 @@ public function generateUrlParams(array $options = [], ?string $model = null, ar
         ) {
             $options['sort'] = $options['direction'] = null;
         }
-
+        $baseUrl = $this->_config['options']['url'] ?? [];
         if (!empty($paging['scope'])) {
             $scope = $paging['scope'];
-            $currentParams = $this->_config['options']['url'];
-
-            if (isset($currentParams['?'][$scope]) && is_array($currentParams['?'][$scope])) {
-                $options += $currentParams['?'][$scope];
+            if (isset($baseUrl['?'][$scope]) && is_array($baseUrl['?'][$scope])) {
+                $options += $baseUrl['?'][$scope];
+                unset($baseUrl['?'][$scope]);
             }
             $options = [$scope => $options];
         }
 
         if (!empty($this->_config['options']['url'])) {
```

Comment:
Should this line also be `!empty($baseUrl)`?

Reply:
@inmun Can you follow up so we can merge or close this?